revoke/revoke.go

```diff
@@ -22,14 +24,85 @@ import (
 	"github.com/cloudflare/cfssl/log"
 )
 
-// HardFail determines whether the failure to check the revocation
-// status of a certificate (i.e. due to network failure) causes
-// verification to fail (a hard failure).
-var HardFail = false
+// Revoke type contains configuration for each new revoke instance
+type Revoke struct {
+	// lock is a Mutex locker for the struct
+	lock sync.Mutex
+	// LocalCRL contains path to the local CRL file. When set, certificate
+	// will be checked only using local CRL script, remote methods will be
+	// skipped.
+	localCRL string
+	// HardFail determines whether the failure to check the revocation
+	// status of a certificate (i.e. due to network failure) causes
+	// verification to fail (a hard failure).
+	hardFail bool
+	// crlSet associates a PKIX certificate list with the URL the CRL is
+	// fetched from.
+	crlSet map[string]*pkix.CertificateList
+}
+
+// defaultChecker is a default config for regular apps which don't need to
+// use custom options.
+var defaultChecker = New(false)
+
+// New creates Revoke config structure.
+// Accepts hardfail bool variable as an option
+func New(hardfail bool) *Revoke {
+	return &Revoke{
+		localCRL: "",
+		hardFail: hardfail,
+		crlSet:   map[string]*pkix.CertificateList{},
+	}
+}
+
+// SetLocalCRL sets localCRL path into the Revoke struct
+func (r *Revoke) SetLocalCRL(localCRLpath string) error {
+	if localCRLpath == "" {
+		r.lock.Lock()
```

Comment:
Yep, it's kinda non-trivial to make it thread safe and also add possibility to reload configuration. Will try to find some time to reimplement this part again.